fenwick/src/main/java/org/opencadc/fenwick/ArtifactSelector.java

```diff
@@ -67,15 +67,35 @@
 
 package org.opencadc.fenwick;
 
-import org.apache.log4j.Logger;
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Iterator;
+
 
 /**
- * 
+ * A Selector to provide the InventoryHarvester with a means to gather include (additive) clauses to select appropriate
+ * Artifacts to be included in the metadata sync merge.
+ *
  * @author pdowler
  */
 public abstract class ArtifactSelector {
-    private static final Logger log = Logger.getLogger(ArtifactSelector.class);
+    final Path selectorConfigDir;
+
+    protected ArtifactSelector() {
+        // Read in the configuration file, and create a usable Map from it.
```

Comment:
Shouldn't all this be in the IncludeArtifacts impl?

If there is nothing  usefully common between ArtifactSelector impls, then this abstract class could be changed to an interface. When I stubbed this out we didn't know if that was the case or not..

Reply:
OK.  Can I make it an interface then?  I thought there may be more implementations that would use this, but I suppose not.